sp.yml

```diff
@@ -0,0 +1,39 @@
+---
+- hosts: "{{ target | default('localhost') }}"
+  become: no
+  connection: ssh
+  gather_facts: true
+
+  vars_files:
+    - "inventory/host_vars/all"
+    - "inventory/group_vars/all"
+    - "inventory/group_vars/{{ env }}/vars.yml"
+    - "inventory/group_vars/{{ env }}/vault.yml"
+    - "inventory/group_vars/{{ pop }}/vars.yml"
+    - "inventory/group_vars/{{ pop }}/vault.yml"
+    - "inventory/group_vars/{{ plat }}/vars.yml"
+    - "inventory/group_vars/{{ plat }}/vault.yml"
+
```

Comment:
you do need facts for multi distro platfoms, otherwise "gather_fact: no" - you know my point of view on this.
NOTE: need to fix "./roles/prepare-env/tasks/docker.yml"
repo: "deb [arch=amd64] https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable""

Reply:
Alberto, what if instead apply to the whole role, we only gather information when you need it with the option
´´´
- name: Gather info
  setup:
´´´

Just before running the task where you need the info? Is it possible?